File: app/master-data-management/src/main/webapp/components/store/storeService.js
//# sourceURL=storeService.js
/**
 * Created by dominik on 04.08.15.
 */

define(["angular"], function (angular) {
    return ["$resource", function ($resource) {

        var storeResource = $resource('/data/stores', {}, {
            deleteStoreByGuid: {
                method: 'DELETE',
                isArray:false,
                url: '/data/stores/guid'
            },
            addStoreToCompany: {
                method: "POST",
                url: "/data/stores/addStoreToCompany",
                isArray: false
            },
            findAllStoreGuids: {
                method: "GET",
                url: "/data/stores/guid",
                isArray: true
            },
            findStoreGuidsByCompanyGuid: {
                method: 'GET',
                url: "/data/stores/companyGuid/:companyGuid",
                isArray: true
            }
        });

        this.findAll = function () {
                var stores = storeResource.query();
            return stores;
        };

        this.save = function(store) {
            return storeResource.save(store);
        }

        this.checkIfResourceInstance = function(instance) {
            return (instance instanceof storeResource);
        };

        this.update = function (store) {
            return store.$save();
        };

        this.save = function(store) {
            return storeResource.save(store);
        };

        this.deleteByGuid = function(guid) {
            return storeResource.deleteStoreByGuid({guid: guid});
        };

        this.addStoreToCompany = function(storeGuid, companyGuid) {
            return storeResource.addStoreToCompany({storeGuid: storeGuid, companyGuid: companyGuid})
        };

        this.findAllStoreGuids = function() {
            return storeResource.findAllStoreGuids();
        };
        this.findStoreGuidsByCompanyGuid = function (companyGuid) {
            return storeResource.findStoreGuidsByCompanyGuid({companyGuid:companyGuid})
        }
    }];
});

File: app/master-data-management/src/main/java/unipos/data/components/company/StoreService.java
package unipos.data.components.company;

import unipos.data.components.company.model.Company;
import unipos.data.components.company.model.Store;

import javax.servlet.http.HttpServletRequest;
import java.util.List;

/**
 * Created by Dominik on 19.12.2015.
 */
public interface StoreService {

    List<Store> findAll();
    List<Store> findByUser(String authToken);

    List<Store> findByUserId(String userGuid);

    Store findByGuid(String guid);
    Store findByDocumentId(String documentId);

    void saveStore(Store store);
    void updateStore(Store store);

    void deleteByGuid(String guid);
    void deleteByDocumentId(String documentId);

    void addStoreToCompany(String storeGuid, String companyGuid);
    void removeStorefromCompany(String storeGuid, String companyGuid);
    void changeStoreCompany(String storeGuid, String oldCompanyGuid, String newCompanyGuid);

    Store getStoreByUserIdAndDeviceId(String userId, String deviceId);
    Store getStoreByUserAndDevice(String authToken, String deviceToken);
    Store getStoreByUserAndDevice(HttpServletRequest request);

    /**
     * This method returns the store, the Unipos-Controller is placed in.
     * A unipos Controller can only get placed in one Store per company.
     * @param companyGuid is the guid of the store the unipos-Cotnroller is placed in
     * @return the Store instance the Controller o is placed in.
     */
    Store getControllerPlacedStore(String companyGuid);

    List<Store> getControllerPlacedStores();

    List<Store> findByCompanyGuid(String companyGuid);
}
